source/index-management/index-management.rst

```diff
@@ -852,6 +852,181 @@ Example::
+  interface DropSearchIndexOptions {}
+
+Standard API for Search Indexes
+-------------------------------
+
+.. code:: typescript
+
+  interface Collection {
+    /**
+    * Convenience method for creating a single search index.
+    * 
+    * @return The name of the created search index
+    * @note Drivers MAY opt to implement this method signature, the signature that
+    *   takes an SearchIndexModel as a parameter, or for those languages with method
+    *   overloading MAY decide to implement both.
+    */
+    createSearchIndex(name: String, definition: Document, options: Optional<CreateSearchIndexOptions>): String;
+
+    /**
+    * Convenience method for creating a single index.
+    * 
+    * @return The name of the created search index
+    * 
+    * @note Drivers MAY opt to implement this method signature, the signature that
+    *   takes an name and a definition as parameters, or for those languages with method
+    *   overloading MAY decide to implement both.
+    */
+    createSearchIndex(model: SearchIndexModel, options: Optional<CreateSearchIndexOptions>): String;
+
+    /**
+    * Creates multiple search indexes on the collection.
+    * 
+    * @return An iterable of the newly created index names.
+    */
+    createSearchIndexes(models: Iterable<SearchIndexModel>, options: CreateSearchIndexOptions): Iterable<String> ;
+
+    /**
+    * Updates the search index with the given name to use the provided 
+    * definition.
+    */
+    updateSearchIndex(name: String, definition: Document, options: Optional<UpdateSearchIndexOptions>): void;
+
+    /**
+    * Drops the search index with the given name.
+    */
+    dropSearchIndex(name: String, options: Optional<DropSearchIndexOptions>): void;
+
+    /**
+    * Gets index information for all search indexes in the collection.
+    */
+    listSearchIndexes(name: Optional<String>, aggregationOptions: Optional<AggregationOptions>, listIndexOptions: Optional<ListSearchIndexOptions>): Cursor<{ name: String }>;
+  }
+
+Index View API for Search Indexes
+---------------------------------
+
+.. code:: typescript
+
+  interface Collection {
+    /**
+     * Returns the search index view for this collection.
+     */
+    searchIndexes(options: Optional<ListSearchIndexesOptions>): SearchIndexView;
+  }
+
+  interface SearchIndexView extends Iterable<Document> {
+    /**
+     * Enumerates the index information for all indexes in the collection. This should be
+     * implemented as described in the :ref:`Enumerate Indexes` section, although the naming
+     * requirement is dropped in favor of the driver language standard for handling iteration
+     * over a sequence of objects.
+     *
+     * @see https://github.com/mongodb/specifications/blob/master/source/enumerate-indexes.rst
+     *
+     * @note For drivers that cannot make the IndexView iterable, they MUST implement a list
+     *   method. See below.
+     */
+    iterator(): Iterator<{ name: string }>;
```

Comment:
Is this going to only return index names or a list of the names and full definitions?